src/lib/raft/raft.c

```diff
@@ -1137,6 +1137,14 @@ raft_promote(struct raft *raft)
 	raft_start_candidate(raft);
 }
 
+void
+raft_resign(struct raft *raft)
```

Comment:
Since you change lib/raft, please, add a unit test for the new function to test/unit/raft.c?

Reply:
Done